module/Olcs/src/Service/Data/IrfoGvPermitType.php

```diff
@@ -70,10 +72,22 @@ public function fetchListData()
             $this->setData('IrfoGvPermitType', false);
 
             if (isset($response->getResult()['results'])) {
-                $this->setData('IrfoGvPermitType', $response->getResult()['results']);
+                $this->setData('IrfoGvPermitType', $this->filterArrayById($response->getResult()['results']));
             }
         }
 
         return $this->getData('IrfoGvPermitType');
     }
+
+    /**
+     * Filter unused IDs out of the array for display (cant remove from DB as 100s of permit records still reference the ID, but not needed for new permits)
+     *
+     * @param array $data Data
+     *
+     * @return array
+     */
+    public function filterArrayById(array $data): array
+    {
+        return array_filter($data, fn($record) => !in_array($record['id'], self::UNUSED_TYPE_IDS));
+    }
```

Comment:
Would this functionality be more appropriate as a database column we filter on? A soft delete of sorts that controls visibility?

Reply:
That was one approach i considered. Making schema changes, probably a new Repository method, changes to the query handler to call that method etc seemed like more to un-pick when the dependent permit records are finally deleted and this isnt needed anymore. Happy to re-factor in that pattern if you think its worthwhile though.

Reply:
I'd say let's refactor to make it easier for us in the future. Decoupling the application from the data is also a positive.

As these agreements usually have a date associated with them, I'd recommend instead of a boolean of "hidden", let's throw a (nullable) date in there instead. Then in the code, if the row has a date populated do a comparison with today. This will enable us to hide these rows on a particular date (expiration of an agreement - which can be done on creation - then we'd not need this type of ticket at all).

Reply:
No problem ill make the updates. :+1: